src/transformers/models/bart/modeling_tf_bart.py

```diff
@@ -937,13 +935,25 @@ def call(
 
             past_key_value = inputs["past_key_values"][idx] if inputs["past_key_values"] is not None else None
 
-            hidden_states, layer_self_attn, present_key_value = decoder_layer(
-                hidden_states,
-                attention_mask=combined_attention_mask,
-                encoder_hidden_states=encoder_hidden_states,
-                encoder_attention_mask=encoder_attention_mask,
-                past_key_value=past_key_value,
-            )
+            def true_fn():
+                return decoder_layer(
+                    hidden_states,
+                    attention_mask=None,
+                    encoder_hidden_states=inputs["encoder_hidden_states"],
+                    encoder_attention_mask=inputs["encoder_attention_mask"],
+                    past_key_value=past_key_value,
+                )
+
+            def false_fn():
```

Comment:
```suggestion
            def compute_result_with_attention():
```
Same, I'm open to any other name that is more descriptive.